lib/puppet/indirector/file_content/rest.rb

```diff
@@ -6,4 +6,34 @@ class Puppet::Indirector::FileContent::Rest < Puppet::Indirector::REST
   desc "Retrieve file contents via a REST HTTP interface."
 
   use_srv_service(:fileserver)
+
+  def find(request)
+    return super unless use_http_client?
+
+    content = StringIO.new
+    content.binmode
+
+    url = URI.parse(Puppet::Util.uri_encode(request.uri))
+    session = Puppet.lookup(:http_session)
+    api = session.route_to(:fileserver, url: url)
+
+    api.get_file_content(
+      path: URI.unescape(url.path),
+      environment: request.environment.to_s,
```

Comment:
Need to pass `links: request.options[:links]`

Reply:
I'm not sure if this is true. If we're managing a link, then it means we have:

```puppet
file { '/path/to/link':
  ensure => link,
  source => 'puppet:///modules/module/path/to/link',
  links => manage,
}
```

In that case, we never try to retrieve the "content" of the link. Instead we get the target/destination of the link from the metadata request.

So if we're retrieving file content, then I don't think `links => :manage` would ever make sense. But I need to double check some things.

Reply:
Given the `filetest` module:

```
[root@fellow-trapper filetest]# tree /etc/puppetlabs/code/environments/production/modules/filetest/
/etc/puppetlabs/code/environments/production/modules/filetest/
├── files
│   ├── link -> target
│   └── target
└── manifests
    └── init.pp
```
If I request the `file_content` for `puppet:///modules/filetest/link`, then puppet returns the target's content (so the link was followed):

```
$ curl -k --key $(bundle exec puppet config print hostprivkey) --cert $(bundle exec puppet config print hostcert) https://fellow-trapper.delivery.puppetlabs.net:8140/puppet/v3/file_content/modules/filetest/link\?environment=production\&links\=manage
link target content
$ curl -k --key $(bundle exec puppet config print hostprivkey) --cert $(bundle exec puppet config print hostcert) https://fellow-trapper.delivery.puppetlabs.net:8140/puppet/v3/file_content/modules/filetest/link\?environment=production\&links\=follow
link target content
$ curl -k --key $(bundle exec puppet config print hostprivkey) --cert $(bundle exec puppet config print hostcert) https://fellow-trapper.delivery.puppetlabs.net:8140/puppet/v3/file_content/modules/filetest/link\?environment=production\&
link target content
```

So the links parameter is meaningless and shouldn't be included in the `get_file_content` API.